addons/l10n_ec_website_sale/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        """ Create invoice(s) for the given Sales Order(s).
+
+        :param bool grouped: if True, invoices are grouped by SO id.
+            If False, invoices are grouped by keys returned by :meth:`_get_invoice_grouping_keys`
+        :param bool final: if True, refunds will be generated if necessary
+        :param date: unused parameter
+        :returns: created invoices
+        :rtype: `account.move` recordset
+        :raises: UserError if one of the orders has no invoiceable lines.
+        """
+        moves = super()._create_invoices(grouped, final, date)
+        for move in moves:
+            if move.transaction_ids:
+                sri_payment_methods = move.transaction_ids.mapped('payment_method_id.l10n_ec_sri_payment_id')
```

Comment:
This seems nice in a way, but maybe the options for MX are not that complicated to use it there. @chklop

Reply:
@jco-odoo you mean that MX is doing this with a different approach?

Reply:
I see, but what if they want to configure the following options:
1. Credit Card
2. Debit Card
3. Other financial instruments (wire transfer)

With the proposed approach they can have the different SRI payment methods in 3 different eCommerce payment methods.

If in the future a new one is added, what will need to be done is create the new SRI Payment method and then associate it with the eCommerce payment method.

<img width="829" alt="image" src="https://github.com/odoo/odoo/assets/45669815/e203ac9b-6d3f-462b-b541-90fa8dc32868">